### ISSUE TITLE
Off by one issue in function styleSelectedText

### ISSUE THREAD

**Issue by jcormier** [davidegiacometti/redmine_shortcuts#14]
selectionEnd was being set 1 greater than expected for ctrl-p, so you would end up with something like below when you started typing.

```
<pre>
asdf</pre>
```

The code should use prepend.length when calculating selectionEnd
```diff
--- view_layouts_base_body_bottom_textile.js.orig       2021-11-15 13:50:15.822980038 -0500
+++ view_layouts_base_body_bottom_textile.js    2021-11-15 13:50:38.928515973 -0500
@@ -11,13 +11,13 @@
         textarea.val($content);
         textarea.focus();
         textarea.prop('selectionStart', $start - prepend.length);
-        textarea.prop('selectionEnd', $end - append.length);
+        textarea.prop('selectionEnd', $end - prepend.length);
     } else {
         $content = $content.slice(0, $start) + prepend + $content.slice($start, $end) + append + $content.slice($end, $content.length);
         textarea.val($content);
         textarea.focus();
         textarea.prop('selectionStart', $start + prepend.length);
-        textarea.prop('selectionEnd', $end + append.length);
+        textarea.prop('selectionEnd', $end + prepend.length);
     }
 }
```

**Comment by davidegiacometti** [comment]
Hi @jcormier 
Nice catch!
This is affecting only textile and looks related to the `</pre>` tag.
I am going to release a bug fix.

**Comment by jcormier** [comment]
While your patch works, I still think it's wrong and very hacky.  I don't see any reason that selectionEnd would depend on the length of the text added after the selection, these offsets are all relative to the start of the text box,  it should only depend on the prepend length as my proposed patch does.

Only the slicing really cares about the length of the append text...

**Comment by davidegiacometti** [comment]
The selected text on the textarea depends on both prepended and appended tags.

- Create a new issue and focus the text area
- Write some text
- Select the text (CTRL + A)
- Add the preformatted tags (CTRL + P)
- The entire text is selected

**Comment by jcormier** [comment]
_Existing Text_ - Selected start = 0, end = 13

After pre

_pre text_ - pre.length = 8
_Existing Text_ - Selected start = 0 + 8, end = 13 + 8
_post text_ - post.length = 9

It doesn't matter how large post text is, the selection start/end positions stay the same

**Comment by jcormier** [comment]
Note: Should have added +1 to pre and post lengths for newlines..., but it doesn't change the math.

**Comment by davidegiacometti** [comment]
@jcormier 
You're absolutely right.. My solution is wrong and your patch is working fine!
I have fixed the 0.5.3 with the correct code.
Thank you for the contribution! 😃
